[feature] Added `CopyableFieldsAdmin` which generalize `UUIDAdmin` cl… (#332)
openwisp/openwisp-utils: README.rst

Without this, when creating new objects, inline items won't be saved
unless users change the default values.

> ``openwisp_utils.admin.UUIDAdmin``

Author: ohk, I'll update that, thanks!

Without this, when creating new objects, inline items won't be saved
unless users change the default values.

``openwisp_utils.admin.CopyableFieldsAdmin``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An admin class that allows us to change admin fields to read-only fields
and provides an improved user experience for copying and pasting the field content.
Useful for auto-generated fields such as UUIDs, secret keys, tokens, etc.

``openwisp_utils.admin.UUIDAdmin``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An admin subclass of ``CopyableFieldsAdmin`` with ``uuid`` as a copyable field, which is kept

> for backward compatibility and convenience, since many models show ``uuid`` as the only copyable field.

Reviewer: Please break each line and maintain a similar line length as the rest of the document, applies also to the section above.

In short, keep line length in check when writing docs, code and code comments.

Author: noted!

Without this, when creating new objects, inline items won't be saved
unless users change the default values.

``openwisp_utils.admin.CopyableFieldsAdmin``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> An admin class that allows us to change admin fields to read-only

Reviewer: ```suggestion
An admin class that allows to set admin fields to be read-only
```

> fields and provides an improved user experience for copying and pasting

Reviewer: ```suggestion
and makes it easy to copy the fields contents.
```

> the field content.

Reviewer: ```suggestion
```

Useful for auto-generated fields such as UUIDs, secret keys, tokens, etc.

``openwisp_utils.admin.UUIDAdmin``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

> An admin subclass of ``CopyableFieldsAdmin`` with ``uuid`` as a

Reviewer: ```suggestion
This class is a subclass of ``CopyableFieldsAdmin`` which sets ``uuid`` as the only
```

> copyable field, which is kept for backward compatibility and convenience,

Reviewer: ```suggestion
copyable field. This class is kept for backward compatibility and convenience,
```

> since many models show ``uuid`` as the only copyable field.

Reviewer: ```suggestion
since different models of various OpenWISP modules show ``uuid`` as the only copyable field.
```
